File: level2/이진_변환_반복하기/solution.js
function solution(s) {
  let cnt = 0;
  let del = 0;
  while (s != "1") {
    cnt++;
    del += (s.match(/0/g) || []).length;
    s = s.replace(/0/g, "").length.toString(2);
  }
  var answer = [cnt, del];
  return answer;
}

console.log(solution("110010101001"));
console.log(solution("01110"));
console.log(solution("1111111"));

File: level2/이진_변환_반복하기/Solution.java
class Solution {
    public int[] solution(String s) {
        int cnt = 0; // 이진 변환 횟수
        int del = 0; // 삭제한 0의 수
        while (!s.equals("1")) {
            StringBuilder sb = new StringBuilder();
            for (char c : s.toCharArray()) {
                if (c == '1')
                    sb.append(c);
                else
                    del++;
            }
            s = Integer.toBinaryString(sb.length());
            cnt++;
        }
        int[] answer = new int[] {cnt, del};
        return answer;
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        System.out.println(Arrays.toString(s.solution("110010101001")));
        System.out.println(Arrays.toString(s.solution("01110")));
        System.out.println(Arrays.toString(s.solution("1111111")));
    }
}
